### PR TITLE
Add message to IndexOutOfBoundsException in InternalNodeModelUtils

### PR DESCRIPTION
**Pull request by tivervac**
An exception without any message is not all that useful

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/InternalNodeModelUtils.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected static LineAndColumn getLineAndColumn(INode anyNode, int documentOffse
 	 */
 	protected static LineAndColumn getLineAndColumn(String text, int[] lineBreaks, int offset) {
 		if (offset > text.length() || offset < 0) {
-			throw new IndexOutOfBoundsException();
+			throw new IndexOutOfBoundsException("Offset (" + offset + ") is longer than the text's length (" + text.length() + ")");
```
Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/InternalNodeModelUtils.java:61]: might miss negative offsets

Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/InternalNodeModelUtils.java:61]: Maybe use `com.google.common.base.Preconditions.checkElementIndex`?

Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/InternalNodeModelUtils.java:61]: Well in fact it would need to be checkPositionIndex.

Comment (tivervac) [reply org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/InternalNodeModelUtils.java]: Done

**Pull request merged**